pypdf/_writer.py

```diff
@@ -2076,7 +2076,7 @@ def add_link(
             target_page_index=page_destination,
             fit=Fit(fit_type=fit, fit_args=args),
         )
-        return self.add_annotation(page_number=pagenum, annotation=annotation)
+        return self.add_annotation(page=pagenum, annotation=annotation)
```

Comment:
This change seems not to be necessary

```suggestion
        return self.add_annotation(page_number=pagenum, annotation=annotation)
```